lib/puppet/type/iis_site.rb

```diff
@@ -20,13 +20,19 @@
     end
 
     newvalue(:present) do
-      provider.create
+      provider.create unless provider.exists?
     end
 
     newvalue(:absent) do
       provider.destroy
     end
 
+    def insync?(is)
+      is.to_s == should.to_s or
```

Comment:
Prefer to use `||` and `&&` for logical or and operations, not their control-flow counterparts. Operator precedence is different.

http://www.virtuouscode.com/2014/08/26/how-to-use-rubys-english-andor-operators-without-going-nuts/